server/sonar-server/src/test/java/org/sonar/server/component/ws/ProjectsDeleteActionTest.java

```diff
@@ -0,0 +1,253 @@
+    userSessionRule.setGlobalPermissions(UserRole.ADMIN);
+    insertNewProjectInDbAndReturnSnapshotId(1);
+    insertNewProjectInDbAndReturnSnapshotId(2);
+    insertNewProjectInDbAndReturnSnapshotId(3);
+    insertNewProjectInDbAndReturnSnapshotId(4);
+
+    ws.newGetRequest("api/projects", "delete")
+      .setParam("keys", "project-key-1, project-key-3, project-key-4").execute();
+    dbSession.commit();
+
+    assertThat(dbClient.componentDao().getByUuids(dbSession, Arrays.asList("project-uuid-1", "project-uuid-3", "project-uuid-4"))).isEmpty();
+    assertThat(dbClient.componentDao().getByUuid(dbSession, "project-uuid-2")).isNotNull();
+  }
+
+  @Test
+  public void delete_documents_indexes() throws Exception {
+    userSessionRule.setGlobalPermissions(UserRole.ADMIN);
+    insertNewProjectInIndexes(1);
+    insertNewProjectInIndexes(2);
+    insertNewProjectInIndexes(3);
+    insertNewProjectInIndexes(4);
+
+    ws.newGetRequest("api/projects", "delete")
+      .setParam("keys", "project-key-1, project-key-3, project-key-4").execute();
+
+    String remainingProjectUuid = "project-uuid-2";
+    assertThat(es.getDocumentFieldValues(IssueIndexDefinition.INDEX, IssueIndexDefinition.TYPE_ISSUE, IssueIndexDefinition.FIELD_ISSUE_PROJECT_UUID))
+      .containsOnly(remainingProjectUuid);
+    assertThat(es.getDocumentFieldValues(IssueIndexDefinition.INDEX, IssueIndexDefinition.TYPE_AUTHORIZATION, IssueIndexDefinition.FIELD_AUTHORIZATION_PROJECT_UUID))
+      .containsOnly(remainingProjectUuid);
+    assertThat(es.getDocumentFieldValues(SourceLineIndexDefinition.INDEX, SourceLineIndexDefinition.TYPE, SourceLineIndexDefinition.FIELD_PROJECT_UUID))
+      .containsOnly(remainingProjectUuid);
+    assertThat(es.getDocumentFieldValues(TestIndexDefinition.INDEX, TestIndexDefinition.TYPE, TestIndexDefinition.FIELD_PROJECT_UUID))
+      .containsOnly(remainingProjectUuid);
+  }
+
+  @Test
+  public void web_service_returns_204() throws Exception {
+    userSessionRule.setGlobalPermissions(UserRole.ADMIN);
+    insertNewProjectInDbAndReturnSnapshotId(1);
+
+    WsTester.Result result = ws.newGetRequest("api/projects", "delete").setParam("uuids", "project-uuid-1").execute();
+
+    assertThat(result.outputAsString().trim()).isEmpty();
+  }
+
+  @Test
+  public void fail_if_insufficient_privileges() throws Exception {
+    userSessionRule.setGlobalPermissions(UserRole.CODEVIEWER, UserRole.ISSUE_ADMIN, UserRole.USER);
+    expectedException.expect(ForbiddenException.class);
+
+    ws.newGetRequest("api/projects", "delete").setParam("uuids", "whatever-the-uuid").execute();
+  }
+
+  @Test
+  public void fail_if_scope_is_not_project() throws Exception {
+    userSessionRule.setGlobalPermissions(UserRole.ADMIN);
+    expectedException.expect(IllegalArgumentException.class);
+    dbClient.componentDao().insert(dbSession, ComponentTesting.newProjectDto("project-uuid").setScope(Scopes.DIRECTORY));
```

Comment:
It would probably be clearer to insert a correct hierarchy here (project -> directory) instead of modifying the scope of the top level project (besides, scope=DIR with qualifier=TRK is not a valid combination IIRC).